test/onnx/test_operators.py

```diff
@@ -1289,6 +1289,10 @@ def forward(self, x):
             input_names=["x"],
             dynamic_axes={"x": {0: "dim_0"}},
         )
+    def test_diagonal(self):
+        x = torch.randn(2, 5, 4, 2)
+        self.assertONNX(lambda x: x.diagonal(offset=-1, dim1=-2, dim2=-1), x)
+
```

Comment:
adding it to test/onnx/test_pytorch_onnx_onnxruntime.py would be better as a numerical validation would be done with the exported model